test/runtests.jl

```diff
@@ -1,125 +1,340 @@
+
+                Impute.dropvars!(df; context=ctx)
+                # The mutating test is broken because we need to making a copy of
+                # the original table
+                @test_broken isequal(df, expected)
+            end
+        end
     end
 
     @testset "Interpolate" begin
-        result = impute(a, :interp; limit=0.2)
+        result = impute(a, Interpolate(; context=ctx))
         @test result == collect(1.0:1.0:20)
-        @test result == interp(a)
+        @test result == interp(a; context=ctx)
+
+        # Test in-place method
+        a2 = copy(a)
+        Impute.interp!(a2; context=ctx)
+        @test a2 == result
 
         # Test interpolation between identical points
         b = ones(Union{Float64, Missing}, 20)
         b[[2, 3, 7]] .= missing
-        @test interp(b) == ones(Union{Float64, Missing}, 20)
+        @test interp(b; context=ctx) == ones(Union{Float64, Missing}, 20)
 
         # Test interpolation at endpoints
         b = ones(Union{Float64, Missing}, 20)
         b[[1, 3, 20]] .= missing
-        result = interp(b)
+        result = interp(b; context=ctx)
         @test ismissing(result[1])
         @test ismissing(result[20])
     end
 
     @testset "Fill" begin
         @testset "Value" begin
             fill_val = -1.0
-            result = impute(a, :fill, fill_val; limit=0.2)
+            result = impute(a, Fill(; value=fill_val, context=ctx))
             expected = copy(a)
             expected[[2, 3, 7]] .= fill_val
 
             @test result == expected
+            @test result == Impute.fill(a; value=fill_val, context=ctx)
         end
 
         @testset "Mean" begin
-            result = impute(a, :fill; limit=0.2)
+            result = impute(a, Fill(; value=mean, context=ctx))
             expected = copy(a)
             expected[[2, 3, 7]] .= mean(a[mask])
 
             @test result == expected
+            @test result == Impute.fill(a; value=mean, context=ctx)
+
+            a2 = copy(a)
+            Impute.fill!(a2; context=ctx)
+            @test a2 == result
         end
     end
 
     @testset "LOCF" begin
-        result = impute(a, :locf; limit=0.2)
+        result = impute(a, LOCF(; context=ctx))
         expected = copy(a)
         expected[2] = 1.0
         expected[3] = 1.0
         expected[7] = 6.0
 
         @test result == expected
+        @test result == Impute.locf(a; context=ctx)
+
+        a2 = copy(a)
+        Impute.locf!(a2; context=ctx)
+        @test a2 == result
     end
 
     @testset "NOCB" begin
-        result = impute(a, :nocb; limit=0.2)
+        result = impute(a, NOCB(; context=ctx))
         expected = copy(a)
         expected[2] = 4.0
         expected[3] = 4.0
         expected[7] = 8.0
 
         @test result == expected
+        @test result == Impute.nocb(a; context=ctx)
+
+        a2 = copy(a)
+        Impute.nocb!(a2; context=ctx)
+        @test a2 == result
     end
 
     @testset "DataFrame" begin
```

Comment:
We  need a similar set of test for at least one other `Table` types.

Since the title of this PR is  Tables.jl API.

Reply:
We also test on a column table, but I don't want a huge set of test dependencies that could produce a lot of false positives.

Reply:
I think a row table even just the standard one from Tables.jl which is just a vector of NamedTuples would be valuable

Reply:
#27